moveit_core/collision_detection_fcl/src/collision_common.cpp

```diff
@@ -589,21 +594,24 @@ bool distanceCallback(fcl::CollisionObjectd* o1, fcl::CollisionObjectd* o2, void
 
         const fcl::Contactd& contact = coll_res.getContact(max_index);
         dist_result.distance = -contact.penetration_depth;
+
 #if (MOVEIT_FCL_VERSION >= FCL_VERSION_CHECK(0, 6, 0))
         dist_result.nearest_points[0] = contact.pos;
         dist_result.nearest_points[1] = contact.pos;
 #else
         dist_result.nearest_points[0] = Eigen::Vector3d(contact.pos.data.vs);
         dist_result.nearest_points[1] = Eigen::Vector3d(contact.pos.data.vs);
-        Eigen::Vector3d normal(contact.normal.data.vs);
 #endif
 
+        thread_local Eigen::Vector3d normal;
         if (cdata->req->enable_nearest_points)
         {
 #if (MOVEIT_FCL_VERSION >= FCL_VERSION_CHECK(0, 6, 0))
-          Eigen::Vector3d normal = contact.normal;
+          normal = contact.normal;
 #else
-          Eigen::Vector3d normal(contact.normal.data.vs);
+          normal[0] = contact.normal.data.vs[0];
+          normal[1] = contact.normal.data.vs[1];
+          normal[2] = contact.normal.data.vs[2];
```

Comment:
That actually doesn't compile:

```
error: no matching function for call to ‘Eigen::Map<Eigen::Matrix<double, 3, 1>, 0, Eigen::Stride<0, 0> >::Map(const double&)’
           normal = Eigen::Map<Eigen::Vector3d>(contact.normal.data.vs[0]);
```
I guess it needs an indicator of (num rows, num columns) and maybe it needs to be converted to Eigen::Matrix first. Overall, this seems more error-prone and I don't have a good way to test it since my FCL version is >0.6.0. I think I'll keep it.

Reply:
Of course, you need to pass the data pointer, not the first element only. Sorry for that detail. I didn't compile this but wrote that out of my head. I don't think, the issue is related to the FCL version. I fixed the comment and added the link to the Eigen::Map doc.

Reply:
Sorry, it still doesn't compile because `contact` is const

Reply:
This should work: `normal = Eigen::Map<const Eigen::Vector3d>(contact.normal.data.vs);` ;)

Reply:
What was wrong with the original code in the first place? Is that a leftover from too many iterations?

Reply:
You are right: the original code was perfectly fine! @AndyZe, could you please restore it for simplicity.

Reply:
If you look at the files changed for these lines I don't think they ended up being changed by this PR.

Reply:
I pushed the requested code simplification: 1b08babd67ae0f2d8b78f73329930b0000bf0a77